Update ci to include code coverage after #1215

casz opened the pull request:
After #1215 the code coverage was not properly calculated.
Let's have jacoco aggregate the report :)

I'd like to use codecov to report diff coverage.

<!--
To mark your pull request as work in progress put the construction emoji 🚧 in your title. (hint: copy it)
Helps us to block accidental merges of unfinished work.
-->

### Your checklist for this pull request

🚨 Please review the [guidelines for contributing](../blob/master/docs/CONTRIBUTING.md) to this repository.

- [x] Make sure you are requesting to **pull a topic/feature/bugfix branch** (right side) and not your master branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or in [Jenkins JIRA](https://issues.jenkins-ci.org)
- [x] Link to relevant pull requests, esp. upstream and downstream changes
- [x] Did you provide a test-case? That demonstrates feature works or fixes the issue.

<!--
Put an `x` into the [ ] to show you have filled the information
-->

oleg-nenashev commented on report/pom.xml:
```diff
@@ -0,0 +1,60 @@
+    <version>${revision}${changelist}</version>
+  </parent>
+
+  <artifactId>report</artifactId>
```
```suggestion
  <artifactId>report-aggregator</artifactId>
```

casz commented:
Coverage went from 80% to 26%
https://app.codacy.com/manual/casz/configuration-as-code-plugin/dashboard

casz commented on report/pom.xml:
```diff
@@ -0,0 +1,60 @@
+      </plugin>
+    </plugins>
+  </build>
+
```
```suggestion
```

casz commented:
Dropped the aggregating report.

GitHub action does not yet have an API, allowing codecov to whitelist github actions coverage reports ie. going tokenless.

So I changed travis to bionic and build with JDK 11 for faster builds.

casz commented:
Hmm test coverage is still not properly calculated.
We need the aggregation

casz commented:
okay plugin-pom's enable-jacoco does not behave well with multi module

casz commented:
lots of derping around :D
